expression/builtin_cast.go

```diff
@@ -1438,7 +1438,7 @@ func (b *builtinCastTimeAsDurationSig) evalDuration(row chunk.Row) (res types.Du
 	if err != nil {
 		return res, false, err
 	}
-	res, err = res.RoundFrac(int8(b.tp.Decimal))
+	res, err = res.RoundFrac(int8(b.tp.Decimal), b.ctx.GetSessionVars().TimeZone)
```

Comment:
It might be possible that `GetSessionVars().TimeZone` is not set if this is used by internal sql. See: https://github.com/pingcap/tidb/issues/25700 for a similar bug.

Is it possible to use Location()